src/ert/scheduler/local_driver.py

```diff
@@ -23,26 +66,81 @@ async def kill(self, iens: int) -> None:
         except KeyError:
             return
 
-    async def _wait_until_finish(
-        self, iens: int, executable: str, /, *args: str, cwd: str
-    ) -> None:
-        proc = await asyncio.create_subprocess_exec(
-            executable,
-            *args,
-            cwd=cwd,
-            preexec_fn=os.setpgrp,
-        )
-
-        if self.event_queue is None:
-            await self.ainit()
-        assert self.event_queue is not None
+    async def _run(self, iens: int, executable: str, /, *args: str, cwd: str) -> None:
+        try:
+            obj: _T = await self.__init(
+                iens,
+                executable,
+                *args,
+                cwd=cwd,
+            )
+        except Exception:
+            await self.event_queue.put((iens, JobEvent.FAILED))
+            return
 
         await self.event_queue.put((iens, JobEvent.STARTED))
         try:
-            if await proc.wait() == 0:
+            if await self.__wait(obj):
                 await self.event_queue.put((iens, JobEvent.COMPLETED))
             else:
                 await self.event_queue.put((iens, JobEvent.FAILED))
         except asyncio.CancelledError:
-            proc.terminate()
+            await self.__kill(obj)
             await self.event_queue.put((iens, JobEvent.ABORTED))
+
+
+class LocalDriver(TaskDriver[Process]):
+    def __init__(self) -> None:
+        super().__init__(LocalDriver._init, LocalDriver._wait, LocalDriver._kill)
+
+    @staticmethod
+    async def _init(iens: int, executable: str, /, *args: str, cwd: str) -> Process:
+        return await asyncio.create_subprocess_exec(
+            executable,
+            *args,
+            cwd=cwd,
+            preexec_fn=os.setpgrp,
+        )
+
+    @staticmethod
+    async def _wait(proc: Process) -> bool:
+        return await proc.wait() == 0
+
+    @staticmethod
+    async def _kill(proc: Process) -> None:
+        proc.terminate()
+        await proc.wait()
+
+
+class MockDriver(TaskDriver[None]):
```

Comment:
Could/should we move this class to the tests? As in `conftest.py` or just at the top of the driver test files?

Reply:
My concern is that this will make us lose type information that may be useful for people writing tests. But I don't really know.

Reply:
We can change name from "Mock" to something else, as it is a "functional" driver, but parametrized. "ParametrizedDriver", "ZeroDriver". Avoiding mock makes it more natural for having it in the source.

Reply:
"EmptyDriver" also popped out from my head.